pkg/signal/trap_linux_test.go

```diff
@@ -0,0 +1,80 @@
+// +build linux
+
+package signal
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTrap(t *testing.T) {
+	sigmap := map[string]os.Signal{
+		"TERM":          syscall.SIGTERM,
+		"QUIT":          syscall.SIGQUIT,
+		"INT":           os.Interrupt,
+		"TERM_MULTIPLE": syscall.SIGTERM,
+		"INT_MULTIPLE":  os.Interrupt,
+	}
+	tmpDir, err := ioutil.TempDir("/tmp", "main")
+	defer os.Remove(tmpDir)
+	require.NoError(t, err)
+	wd, _ := os.Getwd()
+	testHelperCode := wd + "/testfiles/main.go"
+	exePath := tmpDir + "/main"
+	cmd := exec.Command("go", "build", "-o", exePath, testHelperCode)
+	err = cmd.Run()
+	if err != nil {
+		t.Fatal("Error in Running CMD GO Build", err.Error())
+	}
```

Comment:
I think the above few lines could be extracted into a separate function which returns the path to the binary:

```go
func buildTestBinary(t *testing.T, tmpdir string) string {
}
```

also `require.NoError(t, err)` instead of `t.Fatal()`